### PR TITLE
Feat/pymimir cpp tests

### PR DESCRIPTION
**Pull request by maichmueller**
Hi guys,

Summary of the PR:

- Add methods `get_ground_atoms`, `get_static_ground_atoms`, etc. to the bindings and factory c++ class to allow for immediate return of all atoms
- Add a pymimir c++ test executable which allows to run pymimir code from within c++ to test binding code without explicit pip reinstallation. To facilitate this, I also split the pymimir binding into an object library and the module which reuses the object library to avoid recompilation when switching between the targets.
- Added `__str__` methods to the bound vectors that also format the contained elements according to their `__str__` function. This generates more meaningful strings to read (e.g. when using atoms).
- Relies on fmt as a dependency as well (see #11 )

### CODE REVIEW

File: include/mimir/formalism/factories.hpp [hunk 1]
```diff
@@ -348,6 +351,12 @@ GroundAtomList<P> PDDLFactories::get_ground_atoms_from_indices(const Iterable& a
     return result;
 }
 
+template<PredicateCategory P>
```
Comment (drexlerd) [inline include/mimir/formalism/factories.hpp:354]: There is already the function `get_factory` that provides access to all pddl objects, including GroundAtoms. Can we use it instead?

File: python/src/pymimir/bindings.cpp [hunk 2]
```diff
@@ -341,727 +330,682 @@ void init_pymimir(py::module_& m)
...
-             py::overload_cast<const std::vector<size_t>&>(&PDDLFactories::get_ground_atoms_from_indices<Fluent, std::vector<size_t>>, py::const_),
-             py::keep_alive<0, 1>())
-        .def("get_derived_ground_atoms_from_indices",
-             py::overload_cast<const std::vector<size_t>&>(&PDDLFactories::get_ground_atoms_from_indices<Derived, std::vector<size_t>>, py::const_),
-             py::keep_alive<0, 1>())
-        .def("get_object", &PDDLFactories::get_object, py::return_value_policy::reference_internal);
+            .def("get_ground_atoms",
+                 [](py::object py_factory)  // we need an object handle to keep the factory alive for each atom in the list
+                 {
+                     const auto& factory = py::cast<const PDDLFactories&>(py_factory);
```
Comment (drexlerd) [inline python/src/pymimir/bindings.cpp:890]: Indeed, you can do:
```cpp
for (const auto& atom : factory.get_factory<GroundAtomFactory<Static>>()) { 
    // ...
}

```

**Comment by drexlerd** [comment]
Great changes too!

I suggest removing the `get_ground_atoms` function, make use of `get_factory` instead.

It would be great to move the formatting code into the c++ side at some later time to have the nice formatting on both sides.

Can you please also fix the merge conflicts here? Sorry again.

File: python/src/pymimir/bindings.cpp (hunk 2 above)
Comment (maichmueller) [reply python/src/pymimir/bindings.cpp:890]: Good to know! Using the factory also gets rid of the unnecessary intermediate vector allocation

File: include/mimir/formalism/factories.hpp (hunk 1 above)
Comment (maichmueller) [reply include/mimir/formalism/factories.hpp:354]: Sure, do you want the get_ground_atoms() function to be removed on the c++ side altogether or merely change it to use the factory instead?

**Comment by maichmueller** [comment]
Merged the current main branch and resolved the conflicts. I used get_factory where meaningful and changed `get_ground_atoms` to provide a subrange instead of a vector. I would argue it is a meaningful addition to the API to complete the set to get 1 (`get_ground_atom`), multiple (`get_ground_atoms_from_indices`), and now all atoms (`get_ground_atoms`) of a kind.

**Comment by maichmueller** [comment]
Merge-conflicts resolved!

**Pull request merged**